// stock-ticker/js/stock-ticker.js
class StockTicker{
	constructor(element, wide){
		this.main = element;
		this.main.style.overflow = 'hidden';

		this.scrollable = this.main.querySelector('.thing-that-scrolls');
		this.t1 = this.main.querySelector('p.t1');
		this.t2 = this.main.querySelector('p.t2');
		
		this.t1.style.position = this.t2.style.position = 'absolute';

		this.half = this.t1.offsetWidth + 2;

		this.t1.style.position = this.t2.style.position = 'relative';

		this.t1.style.width = this.t2.style.width = this.half+'px';
		this.t1.style.display = this.t2.style.display = 'inline-block';

		
		
		this.t1.style.width = this.t2.style.width = this.half+"px";
		this.main.style.width = wide+"px";
		this.scrollable.style.width = ((this.half*2) + 0) + 'px';
		
		

		


		

		

		var tl = new TimelineMax({repeat:6});
		tl.set(this.main, {opacity:1});
		tl.to(this.scrollable, 5, {x:-this.half, ease:Linear.easeNone});
		tl.set(this.scrollable, {x:0});

	}


}


module.exports = StockTicker;


		// this.t2 = this.t1.cloneNode(true);
		// this.scrollable.appendChild(this.t2);
