// stock-ticker/dist/bundle.js
(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
'use strict';
'strict mode';

var Ticker = require('./stock-ticker.js');

var el = document.getElementById("ticker");
var ticker = new Ticker(el, 299);

},{"./stock-ticker.js":2}],2:[function(require,module,exports){
'use strict';

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var StockTicker = function StockTicker(element, wide) {
	_classCallCheck(this, StockTicker);

	this.main = element;
	this.main.style.overflow = 'hidden';

	this.scrollable = this.main.querySelector('.thing-that-scrolls');
	this.t1 = this.main.querySelector('p.t1');
	this.t2 = this.main.querySelector('p.t2');

	this.t1.style.position = this.t2.style.position = 'absolute';

	this.half = this.t1.offsetWidth + 2;

	this.t1.style.position = this.t2.style.position = 'relative';

	this.t1.style.width = this.t2.style.width = this.half + 'px';
	this.t1.style.display = this.t2.style.display = 'inline-block';

	this.t1.style.width = this.t2.style.width = this.half + "px";
	this.main.style.width = wide + "px";
	this.scrollable.style.width = this.half * 2 + 0 + 'px';

	var tl = new TimelineMax({ repeat: 6 });
	tl.set(this.main, { opacity: 1 });
	tl.to(this.scrollable, 5, { x: -this.half, ease: Linear.easeNone });
	tl.set(this.scrollable, { x: 0 });
};

module.exports = StockTicker;

// this.t2 = this.t1.cloneNode(true);
// this.scrollable.appendChild(this.t2);

},{}]},{},[1]);
